CRM/Contribute/BAO/Contribution.php

```diff
@@ -4732,6 +4728,54 @@ public static function sendMail(&$input, &$ids, $contributionID, &$values, $recu
   }
 
   /**
+   * Generate From email and from name in an array values
+   */
+  public static function generateFromEmailAndName($input, $contribution) {
+    $domain = civicrm_api3('domain', 'getsingle', array('id' => CRM_Core_Config::domainID()));
```

Comment:
So , our first option is that if $input['receipt_from_email'] is passed in then we should not process further & return

array($input['receipt_from_email'], $input['receipt_from_name'])